tests/test_cli/conftest.py

```diff
@@ -31,3 +37,28 @@ def mocked_template_runner() -> TemplateRunner:
         validate_tf_config.return_value = None
 
         yield TemplateRunner()
+
+
+@pytest.fixture(autouse=True)
+def mocked_global_parameters_service(matcha_testing_directory):
+    """Mocked global parameters service.
+
+    Args:
+        matcha_testing_directory (str): Temporary directory for testing.
+
+    Yields:
+        GlobalParameters: GlobalParameters object with mocked properties.
+    """
+    with patch(
+        f"{GLOBAL_PARAMETER_SERVICE_FUNCTION_STUB}.default_config_file_path",
+        new_callable=PropertyMock,
+    ) as file_path, patch(
+        f"{GLOBAL_PARAMETER_SERVICE_FUNCTION_STUB}.user_id",
+        new_callable=PropertyMock,
+    ) as user_id:
+        file_path.return_value = str(
+            os.path.join(str(matcha_testing_directory), ".matcha-ml", "config.yaml")
+        )
+        user_id.return_value = "TestUserID"
+
+        yield GlobalParameters
```

Comment:
Does this need to be an instantiated object?

Reply:
Originally it was instantiated but due to the way the same object is being mocked in the new tests under `tests/test_cli/test_analytics.py` it is required for the object to be uninstantiated.

Reply:
Interesting - what's the reason exactly?

Reply:
I believe it is due to the singleton nature of the class, this means that if the object is instantiated in `conftest.py` if the teardown is not used between every class (basically clearing out the singleton) it will use the same values. The tests within `tests/test_cli/test_analytics.py` checked that a config file isn't created at the start of the test and then checks it has been created after the test. This way the object is not instantiated as soon as the test is run and waits until the class is required and instantiated within the code itself. This is made slightly more confusing with the fact that the tests are done using the `invoke` method and not directly calling functions.